### ISSUE TITLE
Member initialisation and assignment not the exact same thing.

### ISSUE THREAD

**Issue by alexsavio** [UCL/rsd-cppcourse#17]
Hi,

Nice materials! Anyway, I was having a look and found a comment in the section https://github.com/UCL/rsd-cppcourse/blob/master/lessons/03classes.md#defining-constructors.

It says that member initialisation and assignment are the exact same thing. This is not true though. Please have a look at the Item 12 in Effective C++, for example.

Member initialisation through the initialisation list is more efficient and is the correct way of doing it. The member assignment is hiding one step more, which is the initialisation of the string with an empty string. I understand that this is a course for beginners and you might not want to go into deep details, but I wouldn't say they are the exact same thing.

**Comment by jamespjh** [comment]
Agreed. It's a "pedagogical lie" to avoid cognitive load. I like to emphasise the principle of "syntactic sugar" in language design, but I'm overdoing it here. Pull requests welcome...
